build: release Lotus Node v1.33.0-rc1 (#13088)
filecoin-project/lotus: CHANGELOG.md

> * [CHANGELOG_1.1x.md](./documentation/changelog/CHANGELOG_1.1x.md) - v1.10.0 to v1.19.0
> * [CHANGELOG_1.2x.md](./documentation/changelog/CHANGELOG_1.2x.md) - v1.20.0 to v1.29.2

> # Node v1.33.0-rc1 / 2025-05-01

Reviewer: Do we aim to land F3 embedded manifests in this release too? if so I think it deserve s an explicit callout.

Author: Is there an issue or a merged PR for this? The release candidate was created from the master branch this morning. If we intend to add it later and backport it, I suggest we wait to highlight it until it has been merged and backported.

Author: I see we have it listed as a bonus for the RC1 here: https://github.com/filecoin-project/lotus/issues/13004, but it did not make it for the RC1-train, I think we can add it as a dependency for RC2 though if it is not too far away.

> The Lotus v1.33.0-rc1 release introduces experimental v2 APIs with F3 awareness, featuring a new TipSet selection mechanism that significantly enhances how applications interact with the Filecoin blockchain. This release candidate also adds F3-aware Ethereum APIs via the /v2 endpoint while improving existing /v1 APIs and implements intelligent fallback mechanisms between F3 and Expected Consensus.

Reviewer: Worth also calling out that /v2 APIs are also exposed via gateway.

Author: Thanks. Added in: https://github.com/filecoin-project/lotus/pull/13088/commits/575609cfaee50eaeb27d7c3cfdc08aa2f4ad1691

> * [CHANGELOG_1.1x.md](./documentation/changelog/CHANGELOG_1.1x.md) - v1.10.0 to v1.19.0
> * [CHANGELOG_1.2x.md](./documentation/changelog/CHANGELOG_1.2x.md) - v1.20.0 to v1.29.2

# Node v1.33.0-rc1 / 2025-05-01

> The Lotus v1.33.0-rc1 release introduces experimental v2 APIs with F3 awareness, featuring a new TipSet selection mechanism that significantly enhances how applications interact with the Filecoin blockchain. This release candidate also adds F3-aware Ethereum APIs via the /v2 endpoint while improving existing /v1 APIs and implements intelligent fallback mechanisms between F3 and Expected Consensus. These /v2 APIs are also exposed through the Lotus Gateway

Reviewer 2: ```suggestion
The Lotus v1.33.0-rc1 release introduces experimental v2 APIs with F3 awareness, featuring a new TipSet selection mechanism that significantly enhances how applications interact with the Filecoin blockchain. This release candidate also adds F3-aware Ethereum APIs via the /v2 endpoint.  All of the /v2 APIs implement intelligent fallback mechanisms between F3 and Expected Consensus and are exposed through the Lotus Gateway.
```

Removed discussion about /v1 since the changes there are miner.  I don't want someone to think we brought F3 awareness to /v1.

Reviewer 2: Committing this before I make changes on top.

> [!NOTE]
> V2 APIs are highly experimental and subject to change without notice.

> [!IMPORTANT]

> > The `/v2` APIs rely on F3 data even if F3 is not yet finalizing the chain (i.e., `EC.Finalize` is `false` in the live F3 manifest used by all participants). To determine if F3 is actively finalizing, call the `F3GetManifest` API and check if `Manifest.EC.Finalize` is `true`. Only when `EC.Finalize` is `true` will the `/v2` `"finalized"` and `"safe"` tags accurately reflect the chain's finality according to F3. This also applies to Ethereum APIs discussed below.

Reviewer 2: This was moved to https://github.com/filecoin-project/lotus/pull/13088/commits/977d49e559ddf618506504f3337959297d98d5b9

  * Removed internal `Eth*Limited` methods (e.g., `EthGetTransactionByHashLimited`) from the supported gateway API surface.
  * Improved error handling: block selection endpoints now consistently return `ErrNullRound` (and corresponding JSONRPC errors) for null tipsets.

## 🐛 Bug Fix Highlights
- fix(eth): apply limit in EthGetBlockReceiptsLimited ([filecoin-project/lotus#12883](https://github.com/filecoin-project/lotus/pull/12883))
- fix(eth): always return nil for eth transactions not found ([filecoin-project/lotus#12999](https://github.com/filecoin-project/lotus/pull/12999))
- fix(deps): fix Ledger hardware wallet support ([filecoin-project/lotus#13048](https://github.com/filecoin-project/lotus/pull/13048))

## 📝 Changelog

For the full set of changes since the last stable release:

> - Node: https://github.com/filecoin-project/lotus/compare/v1.32.3...v1.33.0-rc1

Reviewer 2: ```suggestion
- Node: https://github.com/filecoin-project/lotus/compare/release/v1.32.3...release/v1.33.0
```

The link didn't work for me otherwise

Reviewer 2: Committing this before I make a proposal on top.
